v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable camelcase */
+const sha256 = require("sha256");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const { EventType } = require("../../../constants");
+
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  flattenJson
+} = require("../../util");
+
+function formatRevenue(revenue) {
+  return Number((revenue || 0).toFixed(2));
+}
+
+function getContentType(message, defaultValue, categoryToContent) {
+  const { options } = message;
+  if (options && options.contentType) {
+    return [options.contentType];
+  }
+
+  let { category } = message.properties;
+  if (!category) {
+    const { products } = message.properties;
+    if (products && products.length) {
+      category = products[0].category;
+    }
+  }
+  if (category) {
+    if (categoryToContent === undefined) {
+      categoryToContent = [];
+    }
+    const mapped = categoryToContent;
+    const mappedTo = mapped.reduce((filtered, map) => {
+      if (map.from === category) {
+        filtered = map.to;
+      }
+      return filtered;
+    }, "");
+    if (mappedTo.length) {
+      return mappedTo;
+    }
+  }
+  return defaultValue;
+}
+
+function handleOrder(message, categoryToContent) {
+  const { products } = message.properties;
+  const value = formatRevenue(message.properties.revenue);
+  const contentType = getContentType(message, "product", categoryToContent);
+  const contentIds = [];
+  const contents = [];
+  const { category } = message.properties;
+
+  for (let i = 0; i < products.length; i += 1) {
+    const pId = products[i].product_id || products[i].sku || products[i].id;
+    contentIds.push(pId);
+    const content = {
+      id: pId,
+      quantity: products[i].quantity,
+      item_price: products[i].price
+    };
+    contents.push(content);
+  }
+  return {
+    content_category: category,
+    content_ids: contentIds,
+    content_type: contentType,
+    currency: message.properties.currency,
+    value,
+    contents,
+    num_items: contentIds.length
+  };
+}
+function handleProductListViewed(message, categoryToContent) {
```

Comment:
add comments explaining the parameters and the use case

Reply:
done